pandas/core/groupby.py

```diff
@@ -1021,6 +1022,45 @@ def cumcount(self, ascending=True):
         cumcounts = self._cumcount_array(ascending=ascending)
         return Series(cumcounts, index)
 
+    def cumprod(self, axis=0):
+        """
+        Cumulative product for each group
+
+        """
+        if axis != 0:
```

Comment:
I would think the axis !=0 can be handled by passing an axis to the transform functions?